src/accelerate/state.py

```diff
@@ -221,6 +222,14 @@ def __init__(
                                 "and/or you do not have an MPS-enabled device on this machine."
                             )
                     else:
+                        from .utils import is_torch_version
+
+                        if not is_torch_version(">", "1.12.0"):
+                            warnings.warn(
+                                "We strongly recommend to install the latest PyTorch nightly version (1.13.0.dev...) on your MacOS machine. "
```

Comment:
That warning may not age well once PyTorch 1.13 is released. Maybe:
```suggestion
                                "We strongly recommend to install PyTorch >= 1.13 (nightly version at the time of writing) on your MacOS machine. "
```